From 9490d69049ad78201787b8b41d42fbb49aa17b70 Mon Sep 17 00:00:00 2001
From: Sofer Athlan-Guyot <sathlang@redhat.com>
Date: Tue, 15 May 2018 12:49:38 +0200
Subject: [PATCH] [Newton Only] Fix getting live value of nova/neutron during update.

In Ieb92ff161d1684c214382c5eb6b5949efc3fe75c we collect the current
value of neutron and nova host parameter.

It appears that we introduced a syntactic bug that was undetected
in the test environment.

This fixes those issues.

Change-Id: I06abb6e0a872319ef958b40ad32384928fa2bc9d
Resolve-Bug: #1771324
---

diff --git a/lib/facter/current_config_hosts.rb b/lib/facter/current_config_hosts.rb
index aee13e4..7da8945 100644
--- a/lib/facter/current_config_hosts.rb
+++ b/lib/facter/current_config_hosts.rb
@@ -1,3 +1,5 @@
+require 'tempfile'
+
 def get_auth(component)
   provider = Object.const_get "Puppet::Provider::#{component.capitalize}"
   auth_func = "#{component}_credentials"
@@ -46,7 +48,7 @@
     File.open(nova_stdin, 'w') do |nova_cmd|
       nova_cmd.puts("import nova.conf\nprint nova.conf.CONF.host")
     end
-    Facter::Core::Execution.execute("nova-manage shell python 2>/dev/null < #{nova_stdin} | sed -e 's/^[> ]*//'")
+    Facter::Core::Execution.execute("nova-manage shell python 2>/dev/null < #{nova_stdin.path} | sed -e 's/^[> ]*//'")
   end
 end
 
